scripts/setup.sh

```diff
@@ -1,31 +1,38 @@
 #!/bin/bash
 
-NODE_VERSION=20.7.0
+echo "Setting up CI environment for Bacon"
 
-# Note: Yarn will automatically switch over to yarn 3 after installing yarn 1.x
-YARN_VERSION=1.22.19
-
-export ORIGINAL_REPO=$REPO
-export CURRENT_DIR=$(pwd)
 REPO=odyssey
 
-cd ${OKTA_HOME}/${REPO}
+REPO_DIR="${OKTA_HOME}/${REPO}"
+
+NODE_VERSION=$(cat "${REPO_DIR}/.nvmrc")
+YARN_VERSION=1.22.19
+
+yum install gnupg2 --allowerasing -y
 
 echo "installing node ${NODE_VERSION}"
+if setup_service node ${NODE_VERSION}; then
+  echo "Installed node ${NODE_VERSION} successfully"
+else
+  echo "Node ${NODE_VERSION} installation failed."
+  exit ${FAILED_SETUP}
+fi
 
-if setup_service node-and-yarn $NODE_VERSION $YARN_VERSION; then
```

Comment:
I'm surprised by this. My understanding was that this version was the preferred method. In most of our repos, we use this: 
```
setup_service node-and-yarn $(cat "${OKTA_HOME}"/"${REPO}"/.nvmrc) "1.22.19"
```
I see you're reading from .nvmrc, so that part is good. For some reason I thought it was preferred to use node-and-yarn 🤷 Take this surprise/concern with a grain of salt; the eng-prod stuff shifts constantly.

Reply:
Kevin added this script, let me check with him on this to verify 🙏

Reply:
cc: @KevinGhadyani-Okta

Reply:
I also thought `node-and-yarn` was the correct way.

I remember the `node-and-yarn` script having issues running `yarn install` on Odyssey. Also, I remember something about it **not** doing a CI-specific install (`--immutable`). So when I swapped over, there was some reason I didn't use it.

The code in that script was pulled from another major repo at Okta: either SIW or Monolith.